### DISCUSSION TITLE
Looking for Java Swagger annotation example that produces proper mocks when imported into microck.

### DISCUSSION THREAD

**Discussion by gsrunion** [microcks/microcks#807] [Q&A] [answered]
As per title, please point me in the direction of a java example. I am having trouble determining which set of swagger annotations need to be present on my endpoints to play nice when imported into Microcks.

**Comment by lbroudoux** [comment, marked as answer]
Hi @gsrunion 

Unfortunately no : we have no sample on how to generate Swagger or OpenAPI contract from annotated Java file. I suspect some users in the community are doing this but mostly as a bootstrap method for getting contract they will later complement with examples and extend.

I suspect that if you want a comprehensive schema that directly worked with Microcks OOTB, you'll have to annotate your Java code with `@ExampleObject` for both request body and responses (see https://javadoc.io/doc/io.swagger.core.v3/swagger-annotations/latest/io/swagger/v3/oas/annotations/media/ExampleObject.html).

Please let us know and share your findings!

**Reply by lbroudoux** [reply]
I've developed such a sample here: https://github.com/lbroudoux/sample-quarkus-microservices/commit/717c5832b9f1b1cc137d1ca7ce052845b21c22d3
